misc/ls-extensions/modules/json-to-record-converter/src/main/java/io/ballerina/converters/JsonToRecordConverter.java

```diff
@@ -0,0 +1,400 @@
+import io.ballerina.compiler.syntax.tree.ArrayTypeDescriptorNode;
+import io.ballerina.compiler.syntax.tree.IdentifierToken;
+import io.ballerina.compiler.syntax.tree.ImportDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModuleMemberDeclarationNode;
+import io.ballerina.compiler.syntax.tree.ModulePartNode;
+import io.ballerina.compiler.syntax.tree.Node;
+import io.ballerina.compiler.syntax.tree.NodeFactory;
+import io.ballerina.compiler.syntax.tree.NodeList;
+import io.ballerina.compiler.syntax.tree.RecordFieldNode;
+import io.ballerina.compiler.syntax.tree.RecordTypeDescriptorNode;
+import io.ballerina.compiler.syntax.tree.SyntaxKind;
+import io.ballerina.compiler.syntax.tree.SyntaxTree;
+import io.ballerina.compiler.syntax.tree.Token;
+import io.ballerina.compiler.syntax.tree.TypeDefinitionNode;
+import io.ballerina.compiler.syntax.tree.TypeDescriptorNode;
+import io.ballerina.converters.exception.ConverterException;
+import io.ballerina.converters.util.Constants;
+import io.ballerina.converters.util.ErrorMessages;
+import io.ballerina.converters.util.SchemaGenerator;
+import io.ballerina.tools.text.TextDocument;
+import io.ballerina.tools.text.TextDocuments;
+import io.swagger.v3.oas.models.Components;
+import io.swagger.v3.oas.models.OpenAPI;
+import io.swagger.v3.oas.models.media.ArraySchema;
+import io.swagger.v3.oas.models.media.ObjectSchema;
+import io.swagger.v3.oas.models.media.Schema;
+import io.swagger.v3.parser.OpenAPIV3Parser;
+import io.swagger.v3.parser.core.models.SwaggerParseResult;
+import org.apache.commons.lang3.StringUtils;
+import org.ballerinalang.formatter.core.Formatter;
+import org.ballerinalang.formatter.core.FormatterException;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Locale;
+import java.util.Map;
+
+import static io.ballerina.compiler.syntax.tree.NodeFactory.createBuiltinSimpleNameReferenceNode;
+import static io.ballerina.converters.util.ConverterUtils.convertOpenAPITypeToBallerina;
+import static io.ballerina.converters.util.ConverterUtils.escapeIdentifier;
+import static io.ballerina.converters.util.ConverterUtils.extractReferenceType;
+
+/**
+ * Implements functionality to convert Json (object or schema) to a ballerina record.
+ *
+ * The conversion is done by using openAPI schema as an intermediary.
+ *
+ * @since 2.0.0
+ */
+public class JsonToRecordConverter {
+
+    private JsonToRecordConverter() {
+        // not called
+    }
+
+    /**
+     * This method takes in a json string and returns the Ballerina record nodes.
+     *
+     * @param jsonString json string for the schema
+     * @return {@link ArrayList<TypeDefinitionNode>} arraylist of type nodes
+     * @throws IOException in case of Json parse error
+     * @throws ConverterException in case of invalid schema
+     */
+    public static String convert(String jsonString) throws IOException,
```

Comment:
Shall we update the Javadoc with the latest changes? Looks like the `@return` and the `FormatterException` need to be updated.

Reply:
Updated in 965ebad